src/modules/analytics/withAnalytics.js

```diff
@@ -1,141 +1,208 @@
 // @flow
 
-import React, { Component } from 'react';
+import React, { Component, type ComponentType, type ElementConfig } from 'react';
 import PropTypes from 'prop-types';
 
-import AnalyticsEvent from './AnalyticsEvent';
-
-const noop = () => { };
-
-export default (WrappedComponent, bindEvents = {}) =>
-  class WithAnalytics extends Component<*> {
-    static defaultProps = {
-      analyticsNamespace: null,
-    };
-
-    static contextTypes = {
-      fireAnalyticsEvent: PropTypes.func,
-      raiseAnalyticsEvent: PropTypes.func,
-      getAnalyticsPath: PropTypes.func,
-    };
-
-    static childContextTypes = {
-      raiseAnalyticsEvent: PropTypes.func,
-      getAnalyticsPath: PropTypes.func,
-    };
-
-    enqueuedEvents = []
-
-    getChildContext = () => ({
-      raiseAnalyticsEvent: this.raiseAnalyticsEvent,
-      getAnalyticsPath: this.getAnalyticsPath,
-    })
-
-    getAnalyticsPath = () => {
-      const { analyticsNamespace } = this.props;
-      const ancestorNamespaces =
-        (typeof this.context.getAnalyticsPath === 'function'
-          && this.context.getAnalyticsPath())
-        || [];
-      return analyticsNamespace
-        ? [...ancestorNamespaces, analyticsNamespace]
-        : ancestorNamespaces;
-    }
-
-    createAnalyticsEvent = (name, payload = {}) => {
-      const meta = { path: this.getAnalyticsPath() };
-      const fire = this.context.fireAnalyticsEvent || noop;
-      const raise = this.raiseAnalyticsEvent;
-      return new AnalyticsEvent(name, payload, meta, { fire, raise });
-    }
-
-    raiseAnalyticsEvent = event => {
-      // If this event is bound to a prop callback, defer it and pass it in when
-      // that callback is fired.
-      if (Object.keys(bindEvents).includes(event.name)) {
-        this.enqueueEvent(event);
-        return;
+import AnalyticsEvent, { type RaiseAnalyticsEvent } from './AnalyticsEvent';
+
+const noop = (...args: any) => {};
```

Comment:
`void`?

Reply:
I've changed to `undefined`, can you return `void` from a function or is it a type only thing?

Reply:
oh derp, no sorry this is correct. my head was in type mode still.